js/components/scrubberBar.js

```diff
@@ -206,11 +234,46 @@ var ScrubberBar = React.createClass({
       playedIndicatorStyle.backgroundColor = this.props.skinConfig.controlBar.adScrubberBar.playedColor;
     }
 
+    if (this.props.controller.state.thumbnails && (this.state.scrubbingPlayheadX || this.lastScrubX || this.state.hoveringX)) {
+      if (this.state.scrubbingPlayheadX) {
+        var hoverPosition = this.state.scrubbingPlayheadX;
+        var hoverTime = (this.state.scrubbingPlayheadX / this.state.scrubberBarWidth) * this.props.duration;
+        playheadClassName += " oo-playhead-scrubbing";
+      }
+      else if (this.lastScrubX) {//to show thumbnail when clicking on playhead
+        var hoverPosition = this.props.currentPlayhead * this.state.scrubberBarWidth / this.props.duration;
+        var hoverTime = this.props.currentPlayhead;
+        playheadClassName += " oo-playhead-scrubbing";
+      }
+      else if (this.state.hoveringX) {
+        var hoverPosition = this.state.hoveringX;
+        var hoverTime=(this.state.hoveringX / this.state.scrubberBarWidth) * this.props.duration;
+        var hoveredIndicatorStyle = {
+              width: Math.min((parseFloat(hoverTime) / parseFloat(this.props.duration)) * 100, 100) + "%",
+              backgroundColor: this.props.skinConfig.controlBar.scrubberBar.playedColor
+            };
+        scrubberBarClassName += " oo-scrubber-bar-hover";
+        playheadClassName += " oo-playhead-hovering";
+      }
+      var thumbnailContainer =
+        (<div className="oo-scrubber-thumbnail-container">
+          <Thumbnail
+            thumbnails={this.props.controller.state.thumbnails}
+            hoverPosition={hoverPosition}
+            duration={this.props.duration}
+            hoverTime={hoverTime > 0 ? hoverTime : 0}
+            scrubberBarWidth={this.state.scrubberBarWidth}/>
+        </div>);
+    }
+    else var thumbnailContainer = null;
```

Comment:
I would just move this to above the if statement without the else, let it be the default. Then you don't need this awkward single line else at the end.